### DISCUSSION TITLE
IM Display not correctly displaying SVG image

### DISCUSSION THREAD

**Discussion by prino** [ImageMagick/ImageMagick#7387] [Development]
### ImageMagick version

7.1.1-29

### Operating system, version and so on

W7-64 Pro

### Description

The attached file, prins1.svg is not displayed correctly in the viewer, but is correctly displayed in 

- Firefox on Windows
- Edge on Windows
- Chrome on Android

And in Inkscape.

Same results for all other prins*.svg files in https://prino.neocities.org/www/sylvain_viard/prins-*.svg (*=1..6)

![prins-1](https://github.com/ImageMagick/ImageMagick/assets/5156665/a3662f8f-c4a5-4dbd-b0fa-dd59f081b353)

**Comment by fmw42** [comment]
Do you have Inkscape installed where IM can find it to use?  Add -verbose to your display command and see if it is using Inkscape

**Comment by prino** [comment]
Inkscape is not on the path, but even if I temporarily add it in a CMD
session, the display via

D:\01-lift\02-prino.neocities.org\www\sylvain_viard>"C:\Program
Files\ImageMagick-7.1.1-Q16-HDRI\imdisplay.exe" -verbose prins-1.svg

doesn't change, and I have no clue as to where the output of "-verbose"
goes.

Robert

On Sun, 9 Jun 2024 at 18:42, Fred Weinhaus ***@***.***> wrote:

> Do you have Inkscape installed where IM can find it to use? Add -verbose
> to your display command and see if it is using Inkscape
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720527>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OPGFXXLMKF2NBICNBTZGSOY7AVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQGUZDO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by fmw42** [reply]
Be sure it is in your $PATH environment variable.  The output should go to your terminal where you issued the command.  Sorry, I do not use Windows and do not know imdisplay.exe.   The IM command in Unix is magick display, but does not work under DOS Windows.  I do not know but expect it would work under Windows Unix.

**Reply by fmw42** [reply]
Zip your SVG file and post it so other Windows users can test with it.

**Comment by prino** [comment]
It's so small that zipping it seems a needless exercise, here it is:

<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg"
     width="6450" height="3780">

 <title>Kilomètres par an 1980-1998</title>

 <defs>
  <g id="d1">
   <rect x="0"   y="0" width="10" height="50"/>
   <rect x="250" y="0" width="10" height="50"/>
   <rect x="500" y="0" width="10" height="50"/>
   <rect x="750" y="0" width="10" height="50"/>
  </g>

  <g id="x" style="fill:#9f9f9f">
   <rect x="0" y="0" width="10" height="10"/>
  </g>

  <g id="y">
   <rect x="0" y="0" width="4810" height="10"/>
  </g>

  <g id="d2" style="fill:#9f9f9f">
   <rect x="0"   y="0"  width="10" height="10"/>
   <rect x="0"   y="70" width="10" height="10"/>
   <rect x="250" y="0"  width="10" height="10"/>
   <rect x="250" y="70" width="10" height="10"/>
  </g>

  <radialGradient id="r" fx="50%" fy="50%" r="60%" spreadMethod="pad">
   <stop offset="0%"   stop-color="#cccccc" stop-opacity="1"/>
   <stop offset="100%" stop-color="#666666" stop-opacity="1"/>
  </radialGradient>
 </defs>

 <rect x="0"    y="0"    width="40"   height="3780"/>
 <rect x="6410" y="0"    width="40"   height="3780"/>
 <rect x="0"    y="0"    width="6450" height="40"  />
 <rect x="0"    y="3740" width="6450" height="40"  />

 <use href="#y" x="920" y="640" />
 <use href="#y" x="920" y="1140"/>
 <use href="#y" x="920" y="1640"/>
 <use href="#y" x="920" y="2140"/>
 <use href="#y" x="920" y="2640"/>
 <use href="#y" x="920" y="3140"/>

 <use href="#d1" x="970"  y="3150"/>
 <use href="#d1" x="1970" y="3150"/>
 <use href="#d1" x="2970" y="3150"/>
 <use href="#d1" x="3970" y="3150"/>
 <use href="#d1" x="4970" y="3150"/>

 <rect style="fill:#9f9f9f" width="10"   height="3700" x="40"   y="40"  />
 <rect style="fill:#9f9f9f" width="10"   height="3700" x="6400" y="40"  />
 <rect style="fill:#9f9f9f" width="6370" height="10"   x="40"   y="40"  />
 <rect style="fill:#9f9f9f" width="6370" height="10"   x="40"   y="3730"/>

 <use href="#x" x="910" y="640" />
 <use href="#x" x="910" y="1140"/>
 <use href="#x" x="910" y="1640"/>
 <use href="#x" x="910" y="2140"/>
 <use href="#x" x="910" y="2640"/>
 <use href="#x" x="910" y="3140"/>

 <use href="#d2" x="970"  y="3130"/>
 <use href="#d2" x="1470" y="3130"/>
 <use href="#d2" x="1970" y="3130"/>
 <use href="#d2" x="2470" y="3130"/>
 <use href="#d2" x="2970" y="3130"/>
 <use href="#d2" x="3470" y="3130"/>
 <use href="#d2" x="3970" y="3130"/>
 <use href="#d2" x="4470" y="3130"/>
 <use href="#d2" x="4970" y="3130"/>
 <use href="#d2" x="5470" y="3130"/>

 <rect width="10" height="2560" x="970"  y="640" />
 <rect width="10" height="2500" x="5720" y="640" />

 <g style="font-size:150px;font-family:sans-serif">
  <text x="2120" y="220" >Kilomètres par an 1980-1998</text>
  <text x="3020" y="3670">Année</text>
 </g>

 <g style="font-size:120px;font-family:sans-serif">
  <text x="450" y="550" >KM</text>
  <text x="450" y="685" >25,000</text>
  <text x="450" y="1185">20,000</text>
  <text x="450" y="1685">15,000</text>
  <text x="450" y="2185">10,000</text>
  <text x="528" y="2685">5,000</text>
  <text x="794" y="3185">0</text>
 </g>

 <g style="font-size:120px;font-family:sans-serif;letter-spacing:-10px">
  <text x="945"  y="3475" transform="rotate(-45  945 3475)">1980</text>
  <text x="1445" y="3475" transform="rotate(-45 1445 3475)">1982</text>
  <text x="1695" y="3475" transform="rotate(-45 1695 3475)">1983</text>
  <text x="2195" y="3475" transform="rotate(-45 2195 3475)">1985</text>
  <text x="2445" y="3475" transform="rotate(-45 2445 3475)">1986</text>
  <text x="2695" y="3475" transform="rotate(-45 2695 3475)">1987</text>
  <text x="2945" y="3475" transform="rotate(-45 2945 3475)">1988</text>
  <text x="3195" y="3475" transform="rotate(-45 3195 3475)">1989</text>
  <text x="3445" y="3475" transform="rotate(-45 3445 3475)">1990</text>
  <text x="3695" y="3475" transform="rotate(-45 3695 3475)">1991</text>
  <text x="3945" y="3475" transform="rotate(-45 3945 3475)">1992</text>
  <text x="4195" y="3475" transform="rotate(-45 4195 3475)">1993</text>
  <text x="4445" y="3475" transform="rotate(-45 4445 3475)">1994</text>
  <text x="4695" y="3475" transform="rotate(-45 4695 3475)">1995</text>
  <text x="4945" y="3475" transform="rotate(-45 4945 3475)">1996</text>
  <text x="5195" y="3475" transform="rotate(-45 5195 3475)">1997</text>
  <text x="5445" y="3475" transform="rotate(-45 5445 3475)">1998</text>
 </g>

 <rect style="fill:url(#r)" width="130" height="298.12"  x="1035"
y="2841.88"><title> 2,981.2 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="790.16"  x="1535"
y="2349.84"><title> 7,901.6 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1202.16" x="1785"
y="1937.84"><title>12,021.6 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="852.99"  x="2285"
y="2287.01"><title> 8,529.9 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="590.37"  x="2535"
y="2549.63"><title> 5,903.7 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="195.17"  x="2785"
y="2944.83"><title> 1,951.7 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="815.28"  x="3035"
y="2324.72"><title> 8,152.8 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="652.81"  x="3285"
y="2487.19"><title> 6,528.1 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1534.63" x="3535"
y="1605.37"><title>15,346.3 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="612.06"  x="3785"
y="2527.94"><title> 6,120.6 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1138.90" x="4035"
y="2001.10"><title>11,389.0 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1318.61" x="4285"
y="1821.39"><title>13,186.1 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1554.29" x="4535"
y="1585.71"><title>15,542.9 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="2159.35" x="4785"
 y="980.65"><title>21,593.5 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="1038.47" x="5035"
y="2101.53"><title>10,384.7 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="454.88"  x="5285"
y="2685.12"><title> 4,548.8 km</title></rect>
 <rect style="fill:url(#r)" width="130" height="539.55"  x="5535"
y="2600.45"><title> 5,395.5 km</title></rect>
</svg>

On Sun, 9 Jun 2024 at 18:56, Fred Weinhaus ***@***.***> wrote:

> Zip your SVG file and post it so other Windows users can test with it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720598>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OK7QXAGPK67F7MFJ2TZGSQOZAVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQGU4TQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by fmw42** [reply]
The reason I asked for it to be zipped was to be sure it did not get changed/distorted when posted.

Nevertheless, I used it to convert to png (magick test.svg x.png) and also used the magick display test.svg and they look the same.

My system has Inkscape 1.3.2 installed and in my $PATH and IM 7.1.1.33 does use it.  

Sorry, I do not know about Windows imdisplay.exe.  Is that provided by Imagemagick or is it a third party display tool?

**Comment by prino** [comment]
imdisplay(.exe) is part of the installation on Windows.

On Sun, 9 Jun 2024 at 19:19, Fred Weinhaus ***@***.***> wrote:

> The reason I asked for it to be zipped was to be sure it did not get
> changed/distorted when posted.
>
> Nevertheless, I used it to convert to png (magick test.svg x.png) and also
> used the magick display test.svg and they look the same.
>
> My system has Inkscape 1.3.2 installed and in my $PATH and IM 7.1.1.33
> does use it.
>
> Sorry, I do not know about Windows imdisplay.exe. Is that provided by
> Imagemagick or is it a third party display tool?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720690>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OOEETWWSTD5RLC725TZGSTD7AVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQGY4TA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by dlemstra** [reply]
It no longer is in the most recent version. I would advise you to use something like [ImageGlass](https://imageglass.org/) instead.

**Comment by fmw42** [comment]
Did you put Inkscape in PATH environment variable?

**Comment by fmw42** [comment]
How is your imdisplay different?   Did you compare to magick.exe test.svg test.png?  Does that look the same?  If not, describe how they are different.

**Comment by prino** [comment]
Yes, I did, and it had no effect.

On Sun, 9 Jun 2024 at 19:42, Fred Weinhaus ***@***.***> wrote:

> Did you put Inkscape in PATH environment variable?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720825>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OITB3MW4II3OZ57Z5DZGSV3LAVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQHAZDK>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Comment by prino** [comment]
Never tried that, but it turns out that "prins-1.png" as converted from
prins-1.svg is just as wrong as the image displayed by imdisplay.

And just opened it with Inkscape, and that image is also wrong, which
deepens the mystery even more, what do FF, Edge, and Chrome cater for that
Inkscape and Magick cannot handle??

On Sun, 9 Jun 2024 at 19:44, Fred Weinhaus ***@***.***> wrote:

> How is your imdisplay different? Did you compare to magick.exe test.svg
> test.png? Does that look the same? If not, describe how they are different.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720831>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OIHUIZT46ESUBDX4J3ZGSWA3AVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQHAZTC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by fmw42** [reply]
I opened your svg in Chrome and saved it as PDF. I then opened the pdf and compared to the png that I created from the svg before and they look the same. I do not see any obvious differences.

Perhaps it is your version of Inkscape or your version of IM.  Try upgrading. I am using IM 7.1.1.33 and Inkscape 1.3.2

Also try a different viewer

**Comment by prino** [comment]
I use W7, have been using Irfanview for probably as long as I can remember,
and am not really interested in "new, and even whiter" interfaces. And I
don't like threads being hijacked.

On Sun, 9 Jun 2024 at 19:55, Dirk Lemstra ***@***.***> wrote:

> It no longer is in the most recent version. I would advise you to use
> something like ImageGlass <https://imageglass.org/> instead.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720905>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OIBBKY6OYTMY6PFCHDZGSXKTAVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQHEYDK>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by fmw42** [reply]
What do you mean "threads being hijacked"?  dlemstra is one of the IM developers and is just trying to be helpful, since I have not been able to be of much help. I am not a developer here, just another user also trying to be helpful.

In this forum, anyone can post an attempt to help.  Advertisers or post with other questions are not permitted on a given question topic.

**Comment by prino** [comment]
I tried upgrading Inkscape a long time ago, I'm on 0.92.4, but had issues,
don't remember what. Will upgrade IM, once I'm back at the other side of
Europe in a few days, during which I will be offline, save for some quick
looks at email.

On Sun, 9 Jun 2024 at 20:05, Fred Weinhaus ***@***.***> wrote:

> I opened your svg in Chrome and saved it as PDF. I then opened the pdf and
> compared to the png that I created from the svg before and they look the
> same. I do not see any obvious differences.
>
> Perhaps it is your version of Inkscape or your version of IM. Try
> upgrading. I am using IM 7.1.1.33 and Inkscape 1.3.2
>
> Also try a different viewer
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/7387#discussioncomment-9720964>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHK6OOJ35GOXFGC5PDTMPDZGSYSJAVCNFSM6AAAAABJBC4ABGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMRQHE3DI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Robert AH Prins
robert(a)prino(d)org
The hitchhiking grandfather <https://prino.neocities.org/index.html>
Some REXX code for use on z/OS
<https://prino.neocities.org/zOS/zOS-Tools.html>

**Reply by fmw42** [reply]
That version is rather old.  Perhaps you should try upgrading it.
